apps/chat/src/components/Marketplace/TabRenderer.tsx

```diff
@@ -128,22 +143,25 @@ export const TabRenderer = ({ screenState }: TabRendererProps) => {
           : true),
     );
 
-    const entitiesForTab =
+    const filteredEntities =
       selectedTab === MarketplaceTabs.MY_APPLICATIONS
-        ? filteredEntities.filter((entity) =>
-            installedModelIds.has(entity.reference),
-          )
-        : filteredEntities;
-
-    const groupedEntities = groupModelsAndSaveOrder(entitiesForTab).slice(
-      0,
-      Number.MAX_SAFE_INTEGER,
-    );
+        ? intersection(entitiesForTab, allEntities)
+        : allEntities;
 
+    const groupedEntities = groupModelsAndSaveOrder(filteredEntities);
     const orderedEntities = groupedEntities.map(
       ({ entities }) => orderBy(entities, 'version', 'desc')[0],
     );
 
+    if (
+      selectedTab === MarketplaceTabs.MY_APPLICATIONS &&
+      orderedEntities.length === 0
+    ) {
+      setSuggestedResults(allEntities);
```

Comment:
do you need all these changes? Why can't we use `filteredEntities` here? if `entitiesForTab` is `0` and `selectedTab` is My workspace